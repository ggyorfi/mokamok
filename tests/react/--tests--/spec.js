import React from 'react';
import Test from '../test';

describe("react", () => {

    it("should render components", () => {
        const node = mokamok.render(<Test text="React" />);
        expect(node.find('>span').text()).to.equal('React');
    });

    it("should handle events with rendered components", () => {
        const stubs = mokamok.stubEventHandlers(Test);
        const node = mokamok.render(<Test text="React" />);
        node.find('>span').simulate('click');
        expect(stubs.onClick).to.be.calledOnce;
        node.find('li:first-child').simulate('blur');
        expect(stubs.onBlur).to.be.calledOnce;
    });

    it("should shallow render components", () => {
        const node = mokamok.shallow(<Test text="React" />);
        expect(node.find('>span').text()).to.equal('React');
    });

    it("should handle events with shallow rendered components", () => {
        const stubs = mokamok.stubEventHandlers(Test);
        const node = mokamok.shallow(<Test text="React" />);
        node.find('>span').simulate('click');
        expect(stubs.onClick).to.be.calledOnce;
        node.find('li:first-child').simulate('blur');
        expect(stubs.onBlur).to.not.be.called;
    });

    it("should add chai jquery", function () {
        const node = mokamok.render(<Test text="React" />);
        expect(node).have.css('background', 'rgb(255, 0, 0)');
        expect(node.find('>a')).have.attr('href', 'http://github.com/ggyorfi/mokamok');
    });

});
