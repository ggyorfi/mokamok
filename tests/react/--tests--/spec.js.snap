exports['react should save snapshot'] = `<div style="background: rgb(255, 0, 0);">
  <a href="http://github.com/ggyorfi/mokamok">Mokamok</a>
  <span>React</span>
  <div>
    <ul>
      <li>A</li>
      <li>B</li>
      <li>C</li>
    </ul>
  </div>
</div>`;
exports['react should save an other snapshot'] = `<div style="background: rgb(255, 0, 0);">
  <a href="http://github.com/ggyorfi/mokamok">Mokamok</a>
  <span>\`React\`</span>
  <div>
    <ul>
      <li>A</li>
      <li>B</li>
      <li>C</li>
    </ul>
  </div>
</div>`;
